### PR TITLE
Gh 90/feature reset password

### LINKED ISSUE #90: 🚀 Feature: create reset password API
### ✒️ What is your desired feature?

Implementing an API that allows users to reset their password. This will involve endpoints for initiating the password reset process and for completing it by setting a new password.

### ❔ Why is this feature needed?

To help users recover their accounts if they forget their password.

### 🏆 How do you aim to achieve this?

I want this feature to:

- Create an endpoint for initiating password reset: This endpoint will accept the user's email and send a reset link.
- Create an endpoint for resetting the password: This endpoint will accept the new password and the token, validate them, and update the user's password.
-  Send a confirmation email: Notify the user once their password has been successfully reset.

### 🔄 Alternative solutions

_No response_

### 🔍 Have you checked to see if this feature has been requested before?

- [X] I checked and didn't find a similar request

### PR DESCRIPTION
**Pull request by preetisharma0123**
### *What* change does this PR introduce?

feat(auth): implement password reset functionality
- Added requestPasswordReset and resetPassword methods in auth-controller.js to handle password reset requests and token-based password updates.
- Updated user model in user.js to include fields for resetPasswordToken and resetPasswordExpires.
- Added new routes in auth-routes.js for initiating and processing password reset requests.

This feature allows users to request a password reset link via email and securely update their password using a token.

test(auth): add test cases for password reset functionality
- Added test cases in auth-controller.test.js to cover the entire password reset flow.
- Mocked nodemailer to simulate email sending in tests.
- Verified various scenarios including valid reset, invalid token, and expired token handling.
These tests ensure the password reset functionality is reliable and handles edge cases correctly.

### *Why* was this change needed?

- Closes #90

### Other information (screenshots)

test

### Checklist

- [X] 📝 I have reviewed my code prior to submission
- [X] ❔ Issue number has been linked (if applicable)
- [X] 🧪 Tests have been added or updated (if applicable)
- [X] 📖 Documentation has been updated (if applicable)
- [X] 📸 Screenshots added (if applicable)

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -13,4 +13,4 @@ fi
 if [ -n "$(git status packages/backend --porcelain --untracked-files=no)" ]; then
   cd packages/backend
   npx lint-staged
-fi
+fiq
```
Comment (xamdoo) [inline .husky/pre-commit]: **issue (blocking):** this seems to be a syntax error (fiq instead of fi), causing all related tests to fail.

File: packages/frontend/package.json [hunk 2]
```diff
@@ -80,7 +80,6 @@
     "eslint-plugin-jest-dom": "^5.1.0",
     "eslint-plugin-storybook": "^0.8.0",
     "eslint-plugin-testing-library": "^6.2.0",
-    "husky": "^9.0.11",
```
Comment (xamdoo) [inline packages/frontend/package.json:83]: **issue (blocking):**  this might affect our pre-commit checks. Could you please clarify why it was removed?

**Review by xamdoo — APPROVED** [review]
